internal/slicer/slicer_test.go

```diff
@@ -515,6 +526,77 @@ var slicerTests = []slicerTest{{
 		"/usr/bin/":      "dir 0755",
 		"/usr/bin/hello": "file 0775 eaf29575",
 	},
+}, {
+	summary: "Custom archives with custom packages",
+	pkgs: map[string]map[string]testPackage{
+		"leptons": {
+			"electron": testPackage{
```

Comment:
Neither in the type definition nor here do we have a hint of what these strings actually mean. This also seems slightly misleading because the testPackage is actually a slice here, isn't it?

We have @rebornplusplus above hinting as this issue too. I'm not sure his suggestion is the best idea either, but it would be nice to make things slightly cleaner. Any suggestions?

Reply:
@niemeyer I've applied @rebornplusplus's suggestion and in the end, I like the result. The definition now looks like:
```
	summary: "Custom archives with custom packages",
	archives: map[string]testArchiveData{
		"leptons": testArchiveData{
			packages: []testPackageData{{
				name: "electron",
				content: testutil.MustMakeDeb([]testutil.TarEntry{
					Dir(0755, "./"),
					Dir(0755, "./mass/"),
```

I've made archive packages a list instead of a map because one archive can in theory contain packages of the same name with different versions and/or architectures. It also reduced one level of indentation.

Does that look better?

The changes are in a new fixup commit. Please let me fixup and force-push before merging.